### DISCUSSION TITLE
FreeCAD's Release Schedule

### DISCUSSION THREAD

**Discussion by sliptonic** [Ondsel-Development/FreeCAD#4] [Blog Topics]
There are a lot of reasons, both technical and historical why FreeCAD's releases come when they do. 

We think FreeCAD is ready for a more predictable and routine release process.  It would be good for end-users as well as developers.
Let's talk about making releases more predictable.

**Comment by adrianinsaval** [comment]
Answering to https://ondsel.com/blog/freecad-unpredictable-release-schedule-hurts-users-and-developers/

I'm not sure if we are ready for two releases a year and scheduled point releases, I think we cna safely shoot for one release a year and 4 point releases, but beyond that I think we are missing manpower, we need more people that will be available to make and test builds on all relevant platforms. Notably we have almost 0 automatization for windows installer generation and conda builds are not straightforward enough since the libs are a moving target.

IMO for a shorter release cycle to be viable we need:

1. Github action to build the windows installer (and .zip file)
2. Libpack equivalent (or something else that would give us predictable dependencies) for macOS and corresponding github action for CI and building .dmg file
3. Optionally a more predictable system to generate AppImages (optional because flatpaks and snaps satisfy the need for predictable dependencies on linux)

for the windows libpack, we should also look into documenting the process of making one from scratch so that it's easier to update a library when we need to. The same applies if we make a libpack for macOS

eventually we need to look into making arm releases for macOS, linux and windows (with that order of priority)

**Reply by sliptonic** [reply]
Thanks for replying. There hasn't been much discussion and it deserves more.  Probably the Ondsel discussion section is the wrong place.  We really don't have a good place in the FreeCAD community for this kind of conceptual / technical discussion.

I agree that two releases a year is more than we can achieve right now.  It should be a goal, not a rule.  

Two per year feels about right in terms of cadence to release new issues so that users aren't tempted to base real work on the development branch.  

Two per year is achievable, but like you said, will require more automation and better process.

For a more predictable system to generate appimages, is that not what conda is supposed to be?

**Reply by adrianinsaval** [reply]
It's supposed to be, but in practice we are chasing new libraries all the time, but thinking about it again I think it's possible to pin dependencies and use the same for all point releases in a series, if so then what we need is to change our practice regarding that and have a conda setup with pinned dependencies, that could be a solution for macOS too.

Btw I didn't realize this was ondsel's discussions! This is something we could put on FreeCAD's discussions I think

**Reply by sliptonic** [reply]
The current configuration of Discussions on the FreeCAD org doesn't have a good category.  We have some maintiner-specific discussions which, I think, are locked to maintainers.  We also have some code-related discussions.  The current configuration encourages the discussion back to the forum for things like this.  
That's why I put the discussion here, but I agree it doesn't belong.
